src/posts/2021-05-03/index.md

```diff
@@ -0,0 +1,662 @@
+  <span style="white-space: nowrap;">(Lorenzo Bianconi,
+  [link](https://lore.kernel.org/bpf/ed670de24f951cfd77590decf0229a0ad7fd12f6.1615201152.git.lorenzo@kernel.org/t/#u))</span>
+
+[Third pull request](https://lore.kernel.org/bpf/20210401233121.65221-1-alexei.starovoitov@gmail.com/):
+
+* Add UDP support to socket maps (“sockmaps”). Only TCP was supported so far.
+  The use case that motivated the change is the need for an efficient solution
+  to proxy connections over AF\_UNIX sockets for thousands of services
+  connected to a daemon, after they have been moved into a VM.
+  <span style="white-space: nowrap;">(Cong Wang,
+  [link](https://lore.kernel.org/bpf/20210331023237.41094-1-xiyou.wangcong@gmail.com/t/#u))</span>
+
+* Improve AF\_XDP selftests and program loading. AF\_XDP sockets need a XDP
+  program to filter the packets to redirect to user space. But when multiple
+  AF\_XDP sockets (“xdpsock” instances) are running on a single interface and
+  one of them is terminated, the XDP program would be automatically unloaded,
+  thus rendering the other sockets unable to operate. Besides improving the
+  selftests, this PR addresses the issue by making libbpf use eBPF “links” to
+  properly reference the XDP programs and make them persistent.
+  <span style="white-space: nowrap;">(Maciej Fijalkowski,
+  [link](https://lore.kernel.org/bpf/20210329224316.17793-1-maciej.fijalkowski@intel.com/t/#u))</span>
+
+* Support calling kernel functions from eBPF programs. This feature has some
+  similarities with eBPF helper functions, which are compiled as part of the
+  kernel and can be called from eBPF programs. But instead of writing dedicated
+  functions, this is about calling pre-existing function from the kernel. This
+  does not apply to _any_ function: kernel functions have to be explicitly
+  marked as allowed for a given eBPF program type for the call to be permitted
+  by the verifier.
+
+  The motivation is to reuse some code portions from the kernel, in particular
+  for those eBPF programs that override specific kernel operations
+  (`BPF_PROG_TYPE_STRUCT_OPS`), like TCP congestion control.
+
+  A crucial difference with eBPF helpers is that kernel functions that can be
+  called are _not_ bounded to a fixed ABI contract. This means that they remain
+  free to evolve, even if this breaks existing eBPF programs.
+
+  The eBPF JIT is required for these calls to work. BTF information is also
+  necessary.
+
+  Several functions directly related to TCP congestion control are
+  [marked as allowed](https://lore.kernel.org/bpf/20210325015124.1543397-1-kafai@fb.com/t/#mbe914ced14b35921b471345b58905f962baf1905)
+  for eBPF programs overriding TCP congestion control.
+  <span style="white-space: nowrap;">(Martin KaFai Lau,
+  [link](https://lore.kernel.org/bpf/20210325015124.1543397-1-kafai@fb.com/t/#u))</span>
+
+* Extend batch map operations (lookup, update, delete) to LPM (_Longest Prefix
+  Match_) maps.
```

Comment:
Because patch set uses “LPM” and pretty much all doc on these maps speak of “LPM” and if I were to keep one, “LPM” would feel more natural (to me). I want readers to relate to what they will see elsewhere. But at the same time, I chose to expand for those who are not familiar with the acronym. Do you really think this is an issue?

Reply:
Ok, makes sense.